parse_grid.py

```
@@ -11,6 +11,19 @@
 
 from utils import log_debug
 
+
+def get_soup(html_content: str) -> BeautifulSoup:
+    """Returns a BeautifulSoup object using lxml with a fallback to html.parser.
+
+    Ensures robust parsing even when system-level lxml dependencies are missing.
+    """
+    try:
+        return BeautifulSoup(html_content, "lxml")
+    except Exception:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)